go-controller/pkg/config/config.go

```diff
@@ -955,6 +997,38 @@ func buildMasterHAConfig(ctx *cli.Context, cli, file *config) error {
 	return nil
 }
 
+func buildHybridOverlayConfig(ctx *cli.Context, cli, file *config) error {
+	// Copy config file values over default values
+	if err := overrideFields(&HybridOverlay, &file.HybridOverlay, &savedHybridOverlay); err != nil {
+		return err
+	}
+
+	// And CLI overrides over config file and default values
+	if err := overrideFields(&HybridOverlay, &cli.HybridOverlay, &savedHybridOverlay); err != nil {
+		return err
+	}
+
+	if HybridOverlay.Enabled {
+		// Since the third address of every cluster subnet is reserved for
+		// the hybrid overlay, only allow enabling it for HostSubnets that
+		// are a /24 or larger.
```

Comment:
This made sense when we were planning to steal that address even for non-overlay-using clusters, but at this point there's no need for it. If the admin wants to do hybrid overlay with tiny subnets, they should be allowed to.